src/bhushan/shop.java

```diff
@@ -0,0 +1,135 @@
+package bhushan;
+class shop{
+	String Maggie,Dosa,oilpouches,panipuri,masala;
+	static int MaggieQ,DosaQ,oilpouchesQ,panipuriQ,masalaQ;
+	
+	void setinitialQ(int Maggie,int Dosa,int oilpouches,int panipuri,int masala){
+		MaggieQ = Maggie;
+		DosaQ = Dosa;
+		oilpouchesQ = oilpouches;
+		panipuriQ = panipuri;
+		masalaQ = masala;
+		
+	}
+	void requestedQ(int reqMaggie,int reqDosa,int reqoilpouches,int reqpanipuri,int reqmasala,boolean pAllowedStatus){
+		if(MaggieQ == 0)
+			System.out.println("Maggie is out of stock");
```

Comment:
Change SOP statement, if customer say 0 packs for maggie it doesn’t mean we don't have Maggie packs.